### PR TITLE
v3-staging: enhance verification script

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Summary

* Adds verification for staging metadata during root signing events
* Clarifies serial number in key verification output
* Adds verification for delegation metadata 
* Adds output for version and expiration of staged/published metadata

```
VERIFYING TUF CLIENT UPDATE

Client successfully initialized, updating and downloading targets...
Client updated to...
	snapshot.json version 25, expires 2022/05/06
	targets.json version 2, expires 2022/05/11
	root.json version 2, expires 2022/05/11
	timestamp.json version 25, expires 2022/05/06
```

or for staging
```
STAGED METADATA

Outputting metadata verification at /home/asraa/git/test-sigstore-root/ceremony/2022-04-02...

Verifying targets.json...
	Success! Signatures valid and threshold achieved
	targets version 2, expires 2022/08/01

Verifying rekor.json...
	Contains 0/1 valid signatures
	rekor version 2, expires 2022/10/01

Verifying revocation.json...
	Contains 0/1 valid signatures
	revocation version 2, expires 2022/10/01

Verifying root.json...
	Success! Signatures valid and threshold achieved
	root version 2, expires 2022/08/01
```

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Review by joshuagl — APPROVED** [review]

**Comment by joshuagl** [comment]
When I build this locally,  I get 

```
# github.com/sigstore/root-signing/cmd/tuf/app
cmd/tuf/app/sign.go:75:13: assignment mismatch: 2 variables but repo.CreateDb returns 3 values
```

which I resolved with:

```
diff --git a/cmd/tuf/app/sign.go b/cmd/tuf/app/sign.go
index 8113b6b..9b23dcc 100644
--- a/cmd/tuf/app/sign.go
+++ b/cmd/tuf/app/sign.go
@@ -72,7 +72,7 @@ func Sign() *ffcli.Command {
 }

 func checkMetaForRole(store tuf.LocalStore, role []string) error {
-       db, err := repo.CreateDb(store)
+       db, _, err := repo.CreateDb(store)
        if err != nil {
                return fmt.Errorf("error creating verification database: %w", err)
        }
```

**Comment by asraa** [comment]
> When I build this locally, I get

Thanks for catching this, it seems like the CI job that tests was missing the build. Added that and fixed.

File: cmd/verify/app/repository.go [hunk 1]
```diff
@@ -99,55 +157,62 @@ func verifyStagedMetadata(repository string) error {
...
 				// we may not have all the sig, allow partial sigs for success
-				log.Printf("\tContains %d/%d valid signatures\n", err.(verify.ErrRoleThreshold).Actual, role.Threshold)
+				log.Printf("\tContains %d/%d valid signatures\n", err.(verify.ErrRoleThreshold).Actual, thresholds[name])
+				_, err := printAndGetSignedMeta(name, signed.Signed)
+				if err != nil {
+					return err
+				}
 			} else if err.Error() == verify.ErrNoSignatures.Error() {
-				log.Printf("\tContains 0/%d valid signatures\n", role.Threshold)
+				log.Printf("\tContains 0/%d valid signatures\n", thresholds[name])
```
Comment (Hayden-IO) [inline cmd/verify/app/repository.go:167]: Can you add a code comment that this intentionally doesn't return an error or something along those lines, just to be clear that this is informational and not a failure?

File: cmd/verify/app/repository.go [hunk 2]
```diff
@@ -99,55 +157,62 @@ func verifyStagedMetadata(repository string) error {
...
 			return nil, err
 		}
 		res[role] = *sm
-		fmt.Printf("\t%s version %d, expires %s\n", role, sm.Version, sm.Expires.Format("2006/01/02"))
 	}
+
 	return res, nil
 }
 
+var (
```
Comment (Hayden-IO) [inline cmd/verify/app/repository.go:210]: where are these populated, by cobra?

**Review by Hayden-IO — APPROVED** [review]

File: cmd/verify/app/repository.go (hunk 2 above)
Comment (asraa) [reply cmd/verify/app/repository.go:210]: yep, they're populated below in the `init` function

File: cmd/verify/app/repository.go (hunk 1 above)
Comment (asraa) [reply cmd/verify/app/repository.go:167]: done!

**Review by dlorenc — APPROVED** [review]

**Pull request merged**